include/sparrow/arrow_array_schema_proxy.hpp

```diff
@@ -80,12 +125,25 @@ namespace sparrow
     private:
 
         std::variant<ArrowArray*, ArrowArray> m_array;
+        ArrowArray* m_array_ptr;
         std::variant<ArrowSchema*, ArrowSchema> m_schema;
+        ArrowSchema* m_schema_ptr;
         std::vector<sparrow::buffer_view<uint8_t>> m_buffers;
         std::vector<arrow_proxy> m_children;
+        std::unique_ptr<arrow_proxy> m_dictionary;
+
+        template <typename AA, typename AS>
+        arrow_proxy(AA&& array, AS&& schema, bool);
```

Comment:
What is that boolean? is it to differentiate that constructor from the public ones? If so consider using a tag type instead, `bool` might be misleading or confusing (in particular at call site).